### PR TITLE
ROTARYIO_MODULE typo; EIC interrupt handler refactor

### PR DESCRIPTION
**Pull request by dhalbert**
This is dependent on https://github.com/adafruit/samd-peripherals/pull/21, so I'll make this a final PR once that's accepted. This draft is being submitted so you can see the changes on the CPy side.

### CODE REVIEW

**Comment by dhalbert** [comment]
Fixes #1656.

File: ports/atmel-samd/eic_handler.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+void shared_eic_handler(uint8_t channel) {
+    uint8_t handler = eic_channel_handler[channel];
+    switch (handler) {
+    case EIC_HANDLER_PULSEIN:
+        pulsein_interrupt_handler(channel);
+        break;
+
+    case EIC_HANDLER_INCREMENTAL_ENCODER:
+        incrementalencoder_interrupt_handler(channel);
+        break;
```
Comment (tannewt) [inline ports/atmel-samd/eic_handler.c]: Want to ifdef protect these so the handler isn't linked in if the module is off?

Comment (dhalbert) [reply ports/atmel-samd/eic_handler.c]: Right, that was the whole point :blush:

**Comment by dhalbert** [comment]
@deshipu I turned off `rotaryio` on `pewpew10` to make its `de_DE` build fit. `rotaryio` was already off by accident; this just turns it off deliberately for that board. If you think you want it, we can try decreasing `CFLAGS_INLINE_LIMIT`. The German translation could probably also be made skinnier.

**Comment by dhalbert** [comment]
@tannewt This is ready to merge. No one else is around; I can wait for you.

**Pull request merged**